common/djangoapps/student/tasks.py

```diff
@@ -0,0 +1,126 @@
+"""
+celery task for course enrollment email
+"""
+import logging  # lint-amnesty, pylint: disable=wrong-import-order
+
+from celery import shared_task
+from django.conf import settings
+from django.contrib.auth.models import User  # lint-amnesty, pylint: disable=imported-auth-user
+from opaque_keys import InvalidKeyError
+from opaque_keys.edx.keys import CourseKey
+
+from common.djangoapps.course_modes.models import CourseMode
+from common.djangoapps.student.helpers import get_course_dates_for_email, get_instructors
+from lms.djangoapps.utils import get_braze_client
+from openedx.core.djangoapps.catalog.utils import get_course_uuid_for_course, get_owners_for_course, \
+    get_course_run_details
+from openedx.core.djangoapps.site_configuration import helpers as configuration_helpers
+from openedx.features.course_experience import ENABLE_COURSE_GOALS
+
+log = logging.getLogger(__name__)
+
+MAX_RETRIES = 1
+COUNTDOWN = 5
+
+
+def _remove_date_key_from_course_dates(course_date):
+    _ = course_date.pop('date')
+    return course_date
+
+
+@shared_task(bind=True, ignore_result=True)
+def send_course_enrollment_email(self, user_id, course_id, course_title, short_description, pacing_type):
+    """
+        Retrieve information about course for sending course enrollment email
+    """
+
+    course_run_fields = [
+        'key', 'title', 'short_description', 'marketing_url', 'pacing_type', 'min_effort',
+        'max_effort', 'weeks_to_complete', 'enrollment_count', 'image', 'staff',
+    ]
+
+    try:
+        user = User.objects.get(id=user_id)
+    except User.DoesNotExist:
+        user = None
```

Comment:
Move this call above `get_course_dates_for_email` and combine the try catch. We only need the user for course dates block. If we can't get the user, we won't be able to make the course dates call.